src/backend/utils/name_validation.c

```diff
@@ -0,0 +1,78 @@
+#include "postgres.h"
```

Comment:
Needs Apache license header.

Please go through file and make sure it conforms to our coding standards -

1. All functions, if/else, for, while, do, statements must have braces - no exceptions.
2. All braces must start on a new line by themselves.
3. No extra spaces at the end of lines. I know, seems dumb, but that's the way we've been doing it.
4. Lines are not to exceed 80 columns. A few characters is okay, if you cannot format it otherwise.
5. When formatting function parameters where the length exceeds 80 characters, align the parameters to each other.
6. If a function is not going to be used externally, make it static.
7. Use defined constants or #defines for constant values. Do not hardcode anything, unless necessary.